esp/src/src-react/components/forms/Login.tsx

```diff
@@ -0,0 +1,197 @@
+import * as Utility from "src/Utility";
+import nlsHPCC from "src/nlsHPCC";
+
+const logger = scopedLogger("src-react/components/forms/Login.tsx");
+
+interface LoginFormValues {
+    username: string;
+    password: string;
+}
+
+const defaultValues: LoginFormValues = {
+    username: "",
+    password: ""
+};
+
+const loginStyles = mergeStyleSets({
+    root: {
+        height: "100%",
+        backgroundColor: "#1A9BD7"
+    },
+    container: {
+        width: "99%",
+        position: "absolute",
+        top: "50%",
+        transform: "translateY(-50%)",
+    },
+    formContainer: {
+        width: "500px",
+        padding: "20px 0",
+        borderRadius: "5px",
+        backgroundColor: "#fff",
+        margin: "auto",
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        "selectors": {
+            "p": {
+                fontSize: "15px"
+            },
+            ".ms-TextField": {
+                width: "300px"
+            }
+        }
+    },
+    button: {
+        fontFamily: "'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, Roboto, 'Helvetica Neue', sans-serif",
+        fontSize: "16px",
+        fontWeight: "600",
+        border: "2px solid #0078d4",
+        cursor: "pointer",
+        padding: "12px 36px",
+        margin: "20px 0 0 0",
+        borderRadius: "2px",
+        color: "white",
+        background: "#1a9bd7",
+        "selectors": {
+            ":hover": {
+                backgroundColor: "#106ebe",
+                border: "2px solid #106ebe",
+                color: "white",
+            }
+        }
+    }
+});
+
+interface LoginProps {
+
+}
+
+export const Login: React.FunctionComponent<LoginProps> = ({
+
+}) => {
+
+    const [userSession, setUserSession, saveUserSession] = useUserSession();
+
+    const { handleSubmit, control } = useForm<LoginFormValues>({ defaultValues });
+
+    const [showError, setShowError] = React.useState(false);
+    const [errorMessage, setErrorMessage] = React.useState("");
+
+    const onSubmit = React.useCallback(() => {
+        handleSubmit(
+            async (data, evt) => {
+                const formData = new FormData();
+
+                for (const key in data) {
+                    formData.append(key, data[key]);
+                }
+
+                const loginResponse = await fetch("/esp/login", {
+                    method: "POST",
+                    body: formData
+                });
+
+                let cookies = Utility.parseCookies();
+
+                if (cookies.ESPAuthenticationMSG && loginResponse?.url.indexOf("esp/files/Login.html") > -1) {
+                    setErrorMessage(cookies.ESPAuthenticationMSG);
+                    setShowError(true);
+                } else {
+                    const authTypeResponse = await fetch("/esp/getauthtype");
+                    const text = await authTypeResponse.text();
```

Comment:
This would be more common:
```
const text = await fetch("/esp/getauthtype").then(response=>{
    return authTypeResponse.text();
}.catch(e=>{
    logger.warn(e.message);
    return "";
})
```